tests/func/test_run.py

```diff
@@ -948,3 +949,31 @@ def test(self):
 
                     mock_run.assert_called_once()
                     mock_checkout.assert_not_called()
+
+
+class TestPersistentOutput(TestDvc):
+    def test_ignore_build_cache(self):
+        warning = "Build cache is ignored when persisting outputs."
+
+        with open("immutable", "w") as fobj:
+            fobj.write("1")
+
+        cmd = [
+            "run",
+            "--overwrite-dvcfile",
+            "--deps",
+            "immutable",
+            "--outs-persist",
+            "greetings",
+            "echo hello>>greetings",
+        ]
+
+        with self._caplog.at_level(logging.WARNING, logger="dvc"):
+            assert main(cmd) == 0
```

Comment:
should we test that that warning is not printed during the first run now? :slightly_smiling_face:

Reply:
sure :+1:

Reply:
done, @efiop  (sorry for such back-and-forth, I should've test it correctly beforehand :sweat_smile:)